ReactNativeClient/root.js

```diff
@@ -660,6 +624,51 @@ class AppComponent extends React.Component {
 		});
 
 		AppState.addEventListener('change', this.onAppStateChange_);
+
+		const sharedData = Platform.OS === 'android' ? await ShareExtension.data() : null;
+		if (sharedData) {
+			this.handleShared(sharedData);
+		}
+	}
+
+	async handleShared(sharedData) {
```

Comment:
Could you move this to an external file, maybe something like root-handleShare.js? It would be a simple function that takes the required dependencies as input - sharedData, props.selectedFolderId, props.dispatch

Doing so would make it easier if some day we decide to refactor this huge root.js file.

Reply:
Will do

Reply:
This is done. I've extracted that function to a separate file (and rewrote in TypeScript). Also extracted away the part that handles permissions -- I plan to reuse it in another PR I have opened